modules/c++/nitf/unittests/test_image_segment_blank_nm_compression.cpp

```diff
@@ -0,0 +1,331 @@
+      {
+         uint8_t *buf = &buffers[testIdx].front() + (ii * bytesPerSegment);
+         nitf::ImageWriter imageWriter = writer.newImageWriter(ii);
+         imageWriter.setWriteCaching(1);
+         nitf::ImageSource iSource;
+         nitf::MemorySource memSource(buf,
+                                      bytesPerSegment,
+                                      0, 1, 0);
+         iSource.addBand(memSource);
+         imageWriter.attachSource(iSource);
+      }
+      writer.write();
+      output_io.close();
+
+      {
+         nitf_Reader *reader=NULL;         /* Reader object */
+         nitf_Record *record = NULL;       /* Record used for input and output */
+         nitf_IOHandle in;            /* Input I/O handle */
+         nitf_ListIterator imgIter;   /* Image segment list iterator */
+         nitf_ListIterator imgEnd;    /* Image segment list iterator */
+         nitf_ImageSegment *seg = NULL; /* Image segment object */
+         nitf_ImageSubheader *subhdr = NULL; /* Image subheader object */
+         nitf_ImageReader *iReader = NULL;   /* Image reader */
+         nitf_BlockingInfo *blkInfo = NULL;  /* Blocking information */
+
+         /*  Image information */
+
+         char imageMode[NITF_IMODE_SZ + 1]; /* Image (blocking) mode */
+
+         /*  Mask structure and mask components */
+
+         nitf_Uint32 imageDataOffset;     /* Offset to actual image data past masks */
+         nitf_Uint32 blockRecordLength;   /* Block mask record length */
+         nitf_Uint32 padRecordLength;     /* Pad mask record length */
+         nitf_Uint32 padPixelValueLength; /* Pad pixel value length in bytes */
+         nitf_Uint8 *padValue;            /* Pad value */
+         nitf_Uint64 *blockMask;          /* Block mask array */
+         nitf_Uint64 *padMask;            /* Pad mask array */
+         size_t imgCtr = 0;
+
+         in = nitf_IOHandle_create(tempNitf.pathname().c_str(),
+                                   NITF_ACCESS_READONLY, NITF_OPEN_EXISTING, &error);
+         TEST_ASSERT_TRUE( (NITF_INVALID_HANDLE(in)!=true) );
+         reader = nitf_Reader_construct(&error);
+         record = nitf_Reader_read(reader, in, &error);
+
+         /* Loop through the image segments */
+         imgIter = nitf_List_begin(record->images);
+         imgEnd = nitf_List_end(record->images);
+         while (nitf_ListIterator_notEqualTo(&imgIter, &imgEnd))
+         {
+            /*  Get information from the image subheader */
+            seg = (nitf_ImageSegment *)nitf_ListIterator_get(&imgIter);
+            TEST_ASSERT_TRUE((seg!=NULL));
+            subhdr = seg->subheader;
+
+            nitf_Field_get(subhdr->imageMode,
+                           imageMode, NITF_CONV_STRING, NITF_IMODE_SZ + 1, &error);
+            imageMode[NITF_IMODE_SZ] = 0;
+
+            /*  Get an image reader which creates the nitf_ImageIO were the masks are */
+
+            iReader = nitf_Reader_newImageReader(reader, imgCtr, NULL, &error);
```

Comment:
Can't you use the C++ layer throughout here and then just use the getter methods to get the raw C pointer when you need it?